src/mame/machine/cxd8442q.h

```diff
@@ -0,0 +1,164 @@
+// license:BSD-3-Clause
+// copyright-holders:Brice Onken
+
+/*
+ * Sony CXD8442Q WSC-FIFOQ APbus FIFO and DMA controller
+ *
+ * The FIFOQ is an APbus DMA controller designed for interfacing some of the simpler and lower speed peripherals
+ * to the APbus while providing DMA capabilities. Each FIFO chip can support up to 4 devices.
+ */
+
+#ifndef MAME_MACHINE_CXD8442Q_H
+#define MAME_MACHINE_CXD8442Q_H
+
+#pragma once
+
+#include "emu.h"
+#include "device.h"
+#include "devfind.h"
+#include "mconfig.h"
```

Comment:
You can’t include the PCH from a header – it causes all sorts of problems.  Also, you’re not supposed to include those separate libemu headers individually due to nasty circular dependencies and stuff.

Reply:
yep, i'm dumb. For some reason, in a few files I didn't include it in the cpp for some reason that eludes me now. Thanks!